File: expectanim/src/main/java/com/github/florent37/expectanim/core/custom/TextColorAnimExpectation.java
package com.github.florent37.expectanim.core.custom;

import android.animation.Animator;
import android.animation.ArgbEvaluator;
import android.animation.ObjectAnimator;
import android.view.View;
import android.widget.TextView;

/**
 * Created by florentchampigny on 21/02/2017.
 */

public class TextColorAnimExpectation extends CustomAnimExpectation {

    private final int textColor;

    public TextColorAnimExpectation(int textColor) {
        this.textColor = textColor;
    }


    @Override
    public Animator getAnimator(View viewToMove) {
        if (viewToMove instanceof TextView) {
            final ObjectAnimator objectAnimator = ObjectAnimator.ofInt(viewToMove, "textColor", ((TextView) viewToMove).getCurrentTextColor(), textColor);
            objectAnimator.setEvaluator(new ArgbEvaluator());
            return objectAnimator;
        } else {
            return null;
        }
    }
}

File: expectanim/src/main/java/com/github/florent37/expectanim/core/custom/TextColorAnimExpectation.kt
package com.github.florent37.expectanim.core.custom

import android.animation.Animator
import android.animation.ArgbEvaluator
import android.animation.ObjectAnimator
import android.animation.ValueAnimator
import android.view.View
import android.widget.TextView

class TextColorAnimExpectation(private val textColor: Int) : CustomAnimExpectation() {

    override fun getAnimator(viewToMove: View): Animator? {
        if (viewToMove is TextView) {
            return ObjectAnimator.ofInt(viewToMove, "textColor", viewToMove.currentTextColor, textColor).apply {
                setEvaluator(ArgbEvaluator())
            }
        } else {
            return null
        }
    }
}
